src/lmrtfy.js

```diff
@@ -161,6 +161,34 @@ class LMRTFY {
             }
         }
     }
+
+    static fromUuid(uuid) {
+        let parts = uuid.split(".");
+        let doc;
+      
+        if (parts.length === 1) return game.actors.get(uuid);
+        // Compendium Documents
+        if ( parts[0] === "Compendium" ) {
+            return undefined;
+        }
+      
+        // World Documents
+        else {
+            const [docName, docId] = parts.slice(0, 2);
+            parts = parts.slice(2);
+            const collection = CONFIG[docName].collection.instance;
+            doc = collection.get(docId);
+        }
+      
+        // Embedded Documents
+        while ( parts.length > 1 ) {
+            const [embeddedName, embeddedId] = parts.slice(0, 2);
+            doc = doc.getEmbeddedDocument(embeddedName, embeddedId);
+            parts = parts.slice(2);
+        }
+        if (doc.actor) doc = doc.actor;
+        return doc || undefined;
```

Comment:
What patterns are you attempting to match in these conditionals? Would a regular expression make more sense?

Reply:
I tend to shy away from regular expression myself, since a lot of people tend to not be able to read the expression. Though, I'm just using the code that Tim from midi-qol proposed (and I assume using in his module too). His suggestions said:

"Sometimes when I want to do a gm LMRTFY roll there are multiple tokens on the scene all with the same actor id (base actor). As of foundry 0.8 there is a way to unambiguously reference the synthetic actor that a token has via token.actor.uuid.

I would like to be able to pass actor uuids to LMRTFY (I create a request packet in my code). To do that a backwards compatible solution would be to define LMRTFY.fromUuid as which is a synchronous version of foundries fromUuid and change the onMessage method from"

It's maybe something we can ask him @tposney? Would a regular expression work better here?